Of course, you can take “**Filtering Data**”, madam/sir. I am preparing a “**Filtering Data**[](http://)” in 5 minutes.

![](https://miro.medium.com/max/1400/0*LLbrg6Gicb6JxQve)

Photo by [Adam Jaime](https://unsplash.com/@arobj?utm_source=medium&utm_medium=referral) on [Unsplash](https://unsplash.com/?utm_source=medium&utm_medium=referral)

##### *Filtering Data is one of the most important topics for Data Scientists. Countless amounts of data are generated every day. On the other hand, not every piece of data is important for your project*.

##### *At this point, You should filter your data!*

In this article, I will explain:
- selecting specific columns
- selecting specific rows and columns
- sorting values of a column
- counting values in column
- finding unique values in column
- finding the number of unique values in the column
- comparing two different columns
- filtering data by values

## **Let’s start!**

Firstly, let’s read our data! If you don’t know how to read data check out [my article](https://medium.com/@ahmettalhabektas/first-step-in-data-world-d2eef7c53e53) or [this notebook](https://www.kaggle.com/code/ahmettalhabektas/first-step-in-data-world) about reading data.

### *Importing Library*

In [1]:
import pandas as pd

### *Reading data*

In [2]:
df=pd.read_csv("../input/titanic/train.csv")
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


# Selecting Specific Columns

> ### *df[“column_name”]*

If you wanna look at just one column, you can use **df[“column_name”]**

#### ***Let’s make an example!***

In [3]:
df["Sex"]

0        male
1      female
2      female
3      female
4        male
        ...  
886      male
887    female
888    female
889      male
890      male
Name: Sex, Length: 891, dtype: object

Or we can use df.column_name; however, the column name shouldn’t include any space.

In [4]:
df.Sex

0        male
1      female
2      female
3      female
4        male
        ...  
886      male
887    female
888    female
889      male
890      male
Name: Sex, Length: 891, dtype: object

#### ***You can use all EDA functions with these functions.***

#### ***Let’s make an example!***

In [5]:
df["Sex"].head()

0      male
1    female
2    female
3    female
4      male
Name: Sex, dtype: object

In [6]:
df.Sex.tail()

886      male
887    female
888    female
889      male
890      male
Name: Sex, dtype: object

> ### *df[[“column_name1”, “column_name2”, “column_name3”]]*

If you want to look at more than one column, you can use **df[[“column_name1”, “column_name2”, “column_name3”]]**. Be careful, there are 2 brackets.

In [7]:
df[["Survived","Sex","Age"]]

,Survived,Sex,Age
0,0,male,22.0
1,1,female,38.0
2,1,female,26.0
3,1,female,35.0
4,0,male,35.0
...,...,...,...
886,0,male,27.0
887,1,female,19.0
888,0,female,NaN
889,1,male,26.0


In [8]:
df[["Survived","Sex","Age"]].head()

,Survived,Sex,Age
0,0,male,22.0
1,1,female,38.0
2,1,female,26.0
3,1,female,35.0
4,0,male,35.0


# Selecting specific rows and columns

> ### *df.loc[x:y,[“column_name1”,“column_name2”]]*

If you want to see specific rows and specific columns, you can use **df.loc[x:y,[“column_name1”, “column_name2”]]** where “x” are “y” are integers.

**Basically**, we are saying “Give me rows from x to y (it is not including y ) and give me columns which are **column_name1** and **column_name2**”

In [9]:
df.loc[0:10,["Age","Sex"]]

,Age,Sex
0,22.0,male
1,38.0,female
2,26.0,female
3,35.0,female
4,35.0,male
5,NaN,male
6,54.0,male
7,2.0,male
8,27.0,female
9,14.0,female


## Other Option

> ### *df.iloc[x:y,z:w]*

**df.iloc** is very similar to df.loc. A small difference is that we write indexes in **df.iloc**; that’s why there is the letter “i” in **df.iloc**.

For looking for specific rows and columns, you can also use **df.iloc[x:y,z:w]** where “x”, “y”, “z” and “w” are integers.

# Sorting values of a column


> ### *df[“”].sort_values()*


If you want to sort values in data from bigger to smaller or smaller to bigger, you can use **df[“”].sort_values()**.

#### **How do we use .sort_values() ?**

First, you should write what you want to sort by.

For instance;

In [10]:
df.sort_values("Age").head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
803,804,1,3,"Thomas, Master. Assad Alexander",male,0.42,0,1,2625,8.5167,NaN,C
755,756,1,2,"Hamalainen, Master. Viljo",male,0.67,1,1,250649,14.5000,NaN,S
644,645,1,3,"Baclini, Miss. Eugenie",female,0.75,2,1,2666,19.2583,NaN,C
469,470,1,3,"Baclini, Miss. Helene Barbara",female,0.75,2,1,2666,19.2583,NaN,C
78,79,1,2,"Caldwell, Master. Alden Gates",male,0.83,0,2,248738,29.0000,NaN,S


Or

In [11]:
df.sort_values(by="Age").head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
803,804,1,3,"Thomas, Master. Assad Alexander",male,0.42,0,1,2625,8.5167,NaN,C
755,756,1,2,"Hamalainen, Master. Viljo",male,0.67,1,1,250649,14.5000,NaN,S
644,645,1,3,"Baclini, Miss. Eugenie",female,0.75,2,1,2666,19.2583,NaN,C
469,470,1,3,"Baclini, Miss. Helene Barbara",female,0.75,2,1,2666,19.2583,NaN,C
78,79,1,2,"Caldwell, Master. Alden Gates",male,0.83,0,2,248738,29.0000,NaN,S


As you can see they sorted from smaller to bigger.

For sorting from bigger to smaller, we use “**ascending=False**”.

#### ***Let’s make an example!***

In [12]:
df.sort_values(by="Age",ascending=False).head(10)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
630,631,1,1,"Barkworth, Mr. Algernon Henry Wilson",male,80.0,0,0,27042,30.0000,A23,S
851,852,0,3,"Svensson, Mr. Johan",male,74.0,0,0,347060,7.7750,NaN,S
493,494,0,1,"Artagaveytia, Mr. Ramon",male,71.0,0,0,PC 17609,49.5042,NaN,C
96,97,0,1,"Goldschmidt, Mr. George B",male,71.0,0,0,PC 17754,34.6542,A5,C
116,117,0,3,"Connors, Mr. Patrick",male,70.5,0,0,370369,7.7500,NaN,Q
672,673,0,2,"Mitchell, Mr. Henry Michael",male,70.0,0,0,C.A. 24580,10.5000,NaN,S
745,746,0,1,"Crosby, Capt. Edward Gifford",male,70.0,1,1,WE/P 5735,71.0000,B22,S
33,34,0,2,"Wheadon, Mr. Edward H",male,66.0,0,0,C.A. 24579,10.5000,NaN,S
54,55,0,1,"Ostby, Mr. Engelhart Cornelius",male,65.0,0,1,113509,61.9792,B30,C
280,281,0,3,"Duane, Mr. Frank",male,65.0,0,0,336439,7.7500,NaN,Q


And now, it is sorted from bigger to smaller.


# Counting values in column

> ### *df[“”].value_counts()*

If you want to see the total counts of values for all rows, you can use **df[“”].value_counts()**.

For example;

In [13]:
df["Sex"].value_counts()

male      577
female    314
Name: Sex, dtype: int64

In [14]:
df["Embarked"].value_counts()

S    644
C    168
Q     77
Name: Embarked, dtype: int64

# Finding unique values in column

> ### *df[“”].unique()*

If you want to see just the unique values which are in the column, you can use **df[“”].unique()**.

An example;

In [15]:
df["Pclass"].unique()

array([3, 1, 2])

We understand that the Pclass column includes 3 values which are 1, 2, and 3.

# Finding the number of unique values in the column

> ### *df[“”].nunique()*

If you want to see just the number of unique values, you can use **df[“”].nunique().**

nunique= number of unique

For instance;

In [16]:
df["PassengerId"].nunique()

891

We understand that We have 891 different ***PassengerId*** so that we have 891 passengers.

# Comparing two different columns

> ### *df.groupby(‘’)[‘’].mean()*

For comparing different columns, you can use **df.groupby(‘’)[‘’].mean()**. However, you can also use statistical methods instead of **mean()**.Just like , **sum()**, **std()**, **min()**, **max()**, **median()** etc.

#### Examples

In [17]:
df.groupby('Sex')['Survived'].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

We can say that 74.2% of females survived and 18.89% of males survived.

In [18]:
df.groupby('Sex')['Survived'].sum()

Sex
female    233
male      109
Name: Survived, dtype: int64

We can say that 233 females survived and 109 males survived.

In [19]:
df.groupby('Sex')['Survived'].std()

Sex
female    0.438211
male      0.391775
Name: Survived, dtype: float64

Std = Standard deviation

In [20]:
df.groupby('Sex')['Fare'].min()

Sex
female    6.75
male      0.00
Name: Fare, dtype: float64

We see that the minimum fare for females is 6.75 and the minimum fare for males is 0.00.

In [21]:
df.groupby('Sex')['Fare'].max()

Sex
female    512.3292
male      512.3292
Name: Fare, dtype: float64

We see that the maximum fare for females is 512.3292 and the maximum fare for males is 512.3292.

In [22]:
df.groupby('Sex')['Age'].median()

Sex
female    27.0
male      29.0
Name: Age, dtype: float64

We see that according to Age, the median of females is 27.0 and the median of males is 29.0.

**Or you can compare a column with more than one column**

In [23]:
df.groupby('Sex')[['Survived','Age']].mean()

,Survived,Age
Sex,,
female,0.742038,27.915709
male,0.188908,30.726645


# Filtering Data by Values

For example, I want to see rows which their Sex is male;

In [24]:
df[df.Sex=='male'].tail(10)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
873,874,0,3,"Vander Cruyssen, Mr. Victor",male,47.0,0,0,345765,9.0000,NaN,S
876,877,0,3,"Gustafsson, Mr. Alfred Ossian",male,20.0,0,0,7534,9.8458,NaN,S
877,878,0,3,"Petroff, Mr. Nedelio",male,19.0,0,0,349212,7.8958,NaN,S
878,879,0,3,"Laleff, Mr. Kristo",male,NaN,0,0,349217,7.8958,NaN,S
881,882,0,3,"Markun, Mr. Johann",male,33.0,0,0,349257,7.8958,NaN,S
883,884,0,2,"Banfield, Mr. Frederick James",male,28.0,0,0,C.A./SOTON 34068,10.5000,NaN,S
884,885,0,3,"Sutehall, Mr. Henry Jr",male,25.0,0,0,SOTON/OQ 392076,7.0500,NaN,S
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.0000,NaN,S
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.0000,C148,C
890,891,0,3,"Dooley, Mr. Patrick",male,32.0,0,0,370376,7.7500,NaN,Q


Or let’s look at females who are bigger than 18 ages;

In [25]:
df[(df.Sex=='female')&(df.Age>=18)].head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
8,9,1,3,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",female,27.0,0,2,347742,11.1333,NaN,S
11,12,1,1,"Bonnell, Miss. Elizabeth",female,58.0,0,0,113783,26.5500,C103,S


Or the mean of males who is bigger than 18 and survived ;

In [26]:
df.Survived[(df.Sex=='male')&(df.Age>=18)].mean()

0.17721518987341772

# Conclusion!

You have learned how to filter data

Congratulations! 🎉🎈🎁

Now you should learn **how is data visualized**.

# Stay tuned!

### The next story will be **“Data Visualisation by matplotlib”**